### PR TITLE
test: kubeflow-pipeline-upgrade-test migrated to GitHub Actions

### PR DESCRIPTION
**Pull request by hbelmiro**
This PR depends on:
- https://github.com/kubeflow/pipelines/pull/10946.

**Description of your changes:**
Resolves: https://github.com/kubeflow/pipelines/issues/10931

PR to remove kubeflow-pipeline-upgrade-test from GoogleCloudPlatform/oss-test-infra

- https://github.com/GoogleCloudPlatform/oss-test-infra/pull/2307

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by hbelmiro** [comment]
cc @chensun @DharmitD

**Comment by rimolive** [comment]
cc @zijianjoy

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: .github/actions/kfp-cluster/action.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        version: v0.22.0
+        node_image: kindest/node:v1.29.2
+
+    - name: Build images
+      shell: bash
+      run: ./scripts/deploy/github/build-images.sh
+
+    - name: Deploy KFP
+      shell: bash
+      run: ./scripts/deploy/github/deploy-kfp.sh
```
Comment (DharmitD) [inline .github/actions/kfp-cluster/action.yml:21]: Looks like this script deploys kfp-tekton and not pipelines. Could you update to add kfp deploy steps here instead?

Comment (hbelmiro) [reply .github/actions/kfp-cluster/action.yml:21]: Fixed in https://github.com/kubeflow/pipelines/pull/10946.
This PR must be rebased and merged after https://github.com/kubeflow/pipelines/pull/10946.

Comment (hbelmiro) [reply .github/actions/kfp-cluster/action.yml:21]: I'm reverting it to draft.

**Comment by hbelmiro** [comment]
Rebased. @DharmitD can you take a second look?
/unhold

**Review by DharmitD — APPROVED** [review]
/lgtm
/Approve

**Pull request merged**